protocols/imap/src/main/java/org/apache/james/imap/encode/ESearchResponseEncoder.java

```diff
@@ -66,10 +63,14 @@ protected void doEncode(ImapMessage acceptableMessage, ImapResponseComposer comp
         if (options.contains(SearchResultOption.COUNT)) {
             composer.message(SearchResultOption.COUNT.name()).message(count);
         }
-        if (all != null && all.length > 0 && options.contains(SearchResultOption.ALL)) {
+        if (!useUid && all != null && all.length > 0 && options.contains(SearchResultOption.ALL)) {
```

Comment:
I can understand that the changes are not easy to read but I tried to summarized in a previous answer : 
- in james core, we changed from Long to MessageUid for message uids.
- imap has the concept of "message sequence number" (msn). Before MessageUid, IdRange was used for the two types : uid and msn. Of course, we don't introduce MessageUid type to use such a poor concept so we introduced UidRange. IdRange is made of uids, IdRange is mande of msn. That's all you have to know.